### PR TITLE
feat: add azure_open_ai_text content provider

### PR DESCRIPTION
**Pull request by anasmuhmd**

### CODE REVIEW

File: mocks/internalpkg/microsoft/azure_open_ai_client.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Code generated by mockery v2.42.1. DO NOT EDIT.
```
Comment (traut) [inline mocks/internalpkg/microsoft/azure_open_ai_client.go:1]: should the spelling of `openai` be fixed in the filename here, to be consistent with the names of the plugins/provider? or is it automatic and expected?

Comment (anasanzari) [reply mocks/internalpkg/microsoft/azure_open_ai_client.go:1]: It is auto generated. It converts pascal case to snake case. We can rename the client to AzureOpenaiClient. Wdyt?

Comment (anasanzari) [reply mocks/internalpkg/microsoft/azure_open_ai_client.go:1]: @traut it is auto generated. Renaming the client to AzureOpenaiClient should fix it.

Comment (anasmuhmd) [reply mocks/internalpkg/microsoft/azure_open_ai_client.go:1]: @traut  it is auto generated. Renaming the client to AzureOpenaiClient should fix it.

**Pull request merged**